server/controllers/posts.js

```diff
@@ -0,0 +1,81 @@
+import db from "../db/db-connection.js";
+
+export const getAllPosts = async (req, res) => {
```

Comment:
I'd use just `getPosts` here, and `getPost` instead of `getSinglePost`.

If your `getPosts` function actually filtered down the posts it fetches, I'd consider having both `getPosts` and `getAllPosts` (and in fact, I'd make `getAllPosts` use `getPosts` without a filter, but now I'm off into the weeds with this comment, ignore me).